centreon/src/Core/HostGroup/Infrastructure/Repository/DbReadHostGroupRepository.php

```diff
@@ -411,7 +411,23 @@ private function retrieveHostGroups(
             'alias' => 'hg.hg_alias',
             'name' => 'hg.hg_name',
             'is_activated' => 'hg.hg_activate',
+            'host_category_id' => 'hc.hc_id',
         ]);
+        if (mb_strpos($requestParameters?->getSearchAsString() ?? '', 'host_category_id')) {
+            $concatenator->appendJoins(
+                <<<'SQL'
+                        LEFT JOIN `:db`.hostgroup_relation hgr
+                            ON hg.hg_id = hgr.hostgroup_hg_id
+                        LEFT JOIN `:db`.host h
+                            ON hgr.host_host_id = h.host_id
+                        LEFT JOIN `:db`.hostcategories_relation hcr
+                            ON h.host_id = hcr.host_host_id
+                        LEFT JOIN `:db`.hostcategories hc
+                            ON hcr.hostcategories_hc_id = hc.hc_id
+                            AND hc.level IS NOT NULL
```

Comment:
⚠️ If you are in an on-prem platform with non-admin user, this request is to open, 
you have to filter hostgroups, host and hostcategories on the user's ACLs

I talked about it with @dmyios, you can see with him for more detailed explanations